src/Dfe.PlanTech.Web/ViewComponents/CategorySectionViewComponent.cs

```diff
@@ -21,26 +28,26 @@ public async Task<IViewComponentResult> InvokeAsync(Category category)
 
     private async Task<CategorySectionViewComponentViewModel> GenerateViewModel(Category category)
     {
-        bool sectionsExist = category.Sections?.Count > 0;
-
-        if (!sectionsExist)
+        if (category.Sections.Count == 0)
         {
             _logger.LogError("Found no sections for category {id}", category.Sys.Id);
 
-            return new CategorySectionViewComponentViewModel()
+            return new CategorySectionViewComponentViewModel
             {
                 NoSectionsErrorRedirectUrl = "ServiceUnavailable"
             };
         }
 
         category = await RetrieveSectionStatuses(category);
 
-        return new CategorySectionViewComponentViewModel()
+        return new CategorySectionViewComponentViewModel
         {
             CompletedSectionCount = category.Completed,
-            TotalSectionCount = category.Sections?.Count ?? 0,
-            CategorySectionDto = GetCategorySectionViewComponentViewModel(category),
-            ProgressRetrievalErrorMessage = category.RetrievalError ? "Unable to retrieve progress, please refresh your browser." : null
+            TotalSectionCount = category.Sections.Count,
+            CategorySectionDto = await GetCategorySectionDto(category).ToListAsync(),
```

Comment:
What happens if this fails?

Reply:
I think the implementations all cover it, but to be safe I've added error handling where the query is called